test/evergreen/tsan_warnings_analysis.py

```
@@ -0,0 +1,66 @@
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+import os
+import re, subprocess
+test_tasks = [
+    "python3 ../test/suite/run.py -j 8 --ignore-stdout",
+    "ctest -j 8 -LE \"long_running\"",
+    "bash test/format/format.sh -j 8 -t 60 rows=10000 ops=50000"
+  ]
+
+# Run the tasks in list.
+task_list = list()
+print(test_tasks)
+for task in test_tasks:
+    try:
+        split_command = task.split()
+        # Tasks are allowed to run with maximum of three hours.
+        subprocess.run(split_command, check=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, timeout=10800)
+    except subprocess.CalledProcessError as exception:
+        print(f'Command {exception.cmd} failed with error {exception.returncode}')
```

Shouldn't we return a non-zero error code in case of some serious issue during the exectuion (like SegFault)?

---

The problem is that if enabling TSAN options slow down test execution and causes some tests to fail because they are time sensitive. Since we care more about the TSAN failures rather than normal failures, I ignore the non-zero error code.

---

I think that we should keep the state when all the tests that we execute under TSAN pass successfully. And that's why I think it is better to return non-zero error code to make evergreen job red and let the person who run this know that something went wrong during the script execution. 

I mean that now we just print error message and return 0 error-code in the end, so if some tests fail one day in the very beginning and that caused reducing the number of TSAN warnings reported we still would see green evergreen job which is not right from my point of view.

---

As discussed offline, the task will run every week.